lib/sles4sap/ipaddr2.pm

```diff
@@ -1915,4 +1946,244 @@ sub ipaddr2_test_other_vm {
     }
 }
 
+=head2 ipaddr2_clean_network_peering
+
+    ipaddr2_clean_network_peering();
+
+Cleanup the network peering if needed.
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_clean_network_peering {
+    return unless get_var("IBSM_RG");
+    my $source_group = ipaddr2_azure_resource_group();
+    my $ibs_mirror_resource_group = get_required_var('IBSM_RG');
+
+    qesap_az_vnet_peering_delete(source_group => $source_group, target_group => $ibs_mirror_resource_group);
+}
+
+=head2 ipaddr2_network_peering
+
+    ipaddr2_network_peering();
+
+Create network peering
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_network_peering {
+    my $ibs_mirror_resource_group = get_required_var('IBSM_RG');
+
+    # remove the older peering
+    my $vnet_name = qesap_az_get_vnet($ibs_mirror_resource_group);
+    qesap_az_clean_old_peerings(rg => $ibs_mirror_resource_group, vnet => $vnet_name);
+
+    my $rg = ipaddr2_azure_resource_group();
+    qesap_az_vnet_peering(source_group => $rg, target_group => $ibs_mirror_resource_group);
+}
+
+=head2 ipaddr2_add_server_repos_to_hosts
+
+    ipaddr2_add_server_repos_to_hosts();
+
+Add download.suse.de server to hosts by specifying IBSM address
+
+=over
+
+=back
+
+=cut
+
+sub ipaddr2_add_server_repos_to_hosts {
+    my $ibsm_ip = get_required_var('IBSM_IP');
+    my $bastion_pubip = ipaddr2_bastion_pubip();
```

Comment:
Please use same pattern used in other place
```
my (%args) = @_;

$args{bastion_ip} //= ipaddr2_bastion_pubip();
```

`ipaddr2_bastion_pubip` is relative expensive operation. The proposed pattern allow test module to calculate it only one and provide it to each lib function that need it.